p2-landing-page/js/app.js

```diff
@@ -22,41 +22,50 @@
  * Define Global Variables
  * 
 */
-
-
-/**
- * End Global Variables
- * Start Helper Functions
- * 
-*/
-
-
-
-/**
- * End Helper Functions
- * Begin Main Functions
- * 
-*/
-
-// build the nav
-
-
-// Add class 'active' to section when near top of viewport
-
-
-// Scroll to anchor ID using scrollTO event
-
-
-/**
- * End Main Functions
- * Begin Events
- * 
-*/
-
-// Build menu 
-
-// Scroll to section on link click
-
-// Set sections as active
-
+let sectionNavs = document.querySelectorAll("section");
+let NavTag = document.getElementById("navbar__list");
+let sectionNavPositions=[];
+let sectionNavLength = sectionNavs.length;
+
+// 1.build the nav get position info array
+sectionNavs.forEach((element,index)=>{
+    let sectionName=element.getAttribute("data-nav");
+    let scrollTarget=element.offsetTop;
+    let liTag=document.createElement('li');
+    liTag.setAttribute("class","nav"+index);
+    liTag.innerHTML=`<a onClick="scrollToSection(${scrollTarget})">${sectionName}</a>`;
+    NavTag.appendChild(liTag);
+})
+
+// 2.Scroll to anchor ID using scrollTO event
+function scrollToSection(sectionID) {
+	window.scrollTo({left:0, top:sectionID, behavior: 'smooth'});
+}
+
+// 3.Add class 'active' to section when near top of viewport
+document.addEventListener("scroll", () => {
+	sectionNavPositions = [];
+	sectionNavs.forEach((element) => sectionNavPositions.push(element.getBoundingClientRect().top+50));
```

Comment:
add a comment explaining why add 50
cc @sebz996

Reply:
modified in newest commit